ACPs/196-precompile-secp256r1/README.md

```diff
@@ -0,0 +1,83 @@
+
+## Motivation
+
+The secp256r1 (P-256) elliptic curve is the standard cryptographic curve used by modern device security systems, including Apple's Secure Enclave, Android Keystore, WebAuthn, and Passkeys. However, Avalanche currently only supports secp256k1 natively, forcing developers to use expensive Solidity-based verification that costs 200k-330k gas per signature verification.
+
+This ACP proposes implementing EIP-7212's secp256r1 precompiled contract to unlock significant ecosystem benefits:
+
+### **Enterprise & Institutional Adoption**
+- **Reduced onboarding friction**: Enterprises can leverage existing biometric authentication infrastructure instead of managing seed phrases or hardware wallets
+- **Regulatory compliance**: Institutions can utilize their approved device security standards and identity management systems
+- **Cost optimization**: 100x gas reduction (from 200k-330k to 3,450 gas) makes enterprise-scale applications economically viable
+
+The 100x gas cost reduction makes these use cases economically viable while maintaining the security properties institutions and users expect from their existing devices.
+
+## Specification
+
+This ACP implements [EIP-7212](https://eips.ethereum.org/EIPS/eip-7212) for secp256r1 signature verification on Avalanche. The specification follows EIP-7212 exactly, with the precompiled contract deployed at address `0x0000000000000000000000000000000000000100`.
+
+### Core Functionality
+- **Input**: 160 bytes (message hash + signature components r,s + public key coordinates x,y)
+- **Output**: 32 bytes (success: `0x...01`, failure: empty)  
+- **Gas Cost**: 3,450 gas (based on EIP-7212 benchmarking)
+- **Validation**: Full compliance with NIST FIPS 186-5 specification
+
+### Activation
+This precompile may be activated as part of Avalanche's next network upgrade. Individual Avalanche L1s and subnets could adopt this enhancement independently through their respective client software updates.
+
+For complete technical specifications, validation requirements, and implementation details, refer to [EIP-7212](https://eips.ethereum.org/EIPS/eip-7212).
+
+## Backwards Compatibility
+
+This ACP introduces a new precompiled contract and does not modify existing functionality. No backwards compatibility issues are expected since:
+
+1. The precompile uses a previously unused address (`0x100`)
+2. No existing opcodes or consensus rules are modified
+3. The change is additive and opt-in for applications
+
+Adoption requires a coordinated network upgrade for the C-Chain. Subnet-EVM chains can adopt this enhancement independently by upgrading their client software.
+
+## Security Considerations
+
+### Cryptographic Security
+- The secp256r1 curve is standardized by NIST and widely vetted
+- Security properties are comparable to secp256k1 (used by ECRECOVER)
+- Implementation follows NIST FIPS 186-5 specification exactly
```

Comment:
> - Implementation follows NIST FIPS 186-5 specification exactly

Which implementation does? [This](https://pkg.go.dev/crypto/elliptic#P256) one?

Reply:
To be accurate, Go's implementation actually follows FIPS 186-3. But the P-256 curve itself is identical across FIPS 186-3, 186-4, and 186-5. The core mathematical parameters, signature verification algorithms, and implementation approach are the same.

I'm specifying we are using the version FIPS 186-3, as that's what the Go implementation uses.